src/collect-table.cc

```diff
@@ -2423,21 +2447,31 @@ static void collection_table_cell_data_cb(GtkTreeViewColumn *, GtkCellRenderer *
 	g_autofree gchar *display_text = nullptr;
 	if (info && info->fd)
 		{
-		if (ct->show_text && ct->show_stars)
+		if (ct->show_text)
 			{
-			display_text = g_strconcat(info->fd->name, "\n", star_rating, NULL);
+			display_text = g_strdup(info->fd->name);
 			}
-		else if (ct->show_text)
+
+		if (ct->show_stars && display_text)
 			{
-			display_text = g_strdup(info->fd->name);
+			gchar *t = g_strconcat(display_text,"\n",star_rating,NULL);
```

Comment:
Please use `g_autofree` and `std::swap`. Also respect formatting, i.e. add space after commas.

Reply:
On second thought consider using GString for `display_text` build.

Reply:
Switched to GString. Is my use of g_autoptr correct?